src/components/OAuth/Google/Google.tsx

```diff
@@ -0,0 +1,108 @@
+import { FC, useEffect, useState } from "react";
+import { useGoogleLogin } from "@react-oauth/google";
+
+import GoogleApi from "./GoogleApi";
+import { useGetUrlParams } from "../hooks";
+import { useAppDispatch } from "@/src/redux/hooks";
+import { SocialsProps, OAuthVariant } from "../OAuth.type";
+import styles from "../OAuth.module.scss";
+
+import { UIbutton } from "../../Buttons";
+import { useNavigate } from "react-router-dom";
+import { Icon, IconEnum } from "../..";
+
+const { VITE_BASE_OAUTH_STATE, VITE_GOOGLE_STATE, VITE_GOOGLE_REDIRECT_URI } =
+  import.meta.env;
+
+const Google: FC<SocialsProps> = ({
+  stateId,
+  oAuthVariant,
+  googlePopupMode = false,
+  text = "Enter with",
+  iconSize = 24,
+  buttonSize = "small",
+  buttonColor = "secondary",
+  buttonVariant = "outlined",
+  dataAutomation = "oAuthButton",
+  className,
+}) => {
+  const { state, code, currentLocation, setSearchParams } = useGetUrlParams();
+  const [googleAuthCode, setGoogleAuthCode] = useState(code);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const dispatch = useAppDispatch();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (
+      oAuthVariant === OAuthVariant.GOOGLE &&
+      googleAuthCode &&
+      (state === VITE_GOOGLE_STATE + stateId ||
+        state === VITE_GOOGLE_STATE + VITE_BASE_OAUTH_STATE)
+    ) {
+      new GoogleApi({
+        token: googleAuthCode,
+        redirectUri: currentLocation,
+        setSearchParams,
+        navigate,
+        setIsLoading,
+        setAuthCode: setGoogleAuthCode,
+        dispatch,
+      }).login();
+    }
+  }, [
+    currentLocation,
```

Comment:
I am not sure that good idea add methods like dependencies

Reply:
why, all these methods that are included in the dependency array are hooks, they will not be recreated on every render?